Tests/TutorLizard.BusinessLogic.Tests/Services/Student/StudentServiceScheduleItemRequestTests.cs

```diff
@@ -0,0 +1,84 @@
+﻿using Moq;
+using TutorLizard.BusinessLogic.Interfaces.Data.Repositories;
+using TutorLizard.BusinessLogic.Models;
+using TutorLizard.BusinessLogic.Models.DTOs.Requests;
+using TutorLizard.BusinessLogic.Services;
+
+namespace TutorLizard.BusinessLogic.Tests.Services.Student
+{   
+    public class StudentServiceScheduleItemRequestTests : StudentServiceTestBase
+    {
+        [Fact]
+        public async Task CreateScheduleItemRequest_WhenIsRemoteIsTrue_ShouldSetIsRemoteToTrue()
+        {
+            // Arrange
+            var scheduleItem = new ScheduleItem { Id = 1, Ad = new Ad { TutorId = 2 } };
+            SetupMockGetScheduleItemById(scheduleItem);
+
+            var request = new CreateScheduleItemRequestRequest
+            {
+                StudentId = 2,
+                ScheduleItemId = 1,
+                IsRemote = true
+            };
+
+            // Act
+            await StudentService.CreateScheduleItemRequest(request);
+
+            // Assert
+            MockScheduleItemRequestRepository.Verify(repo => repo.Create(It.Is<ScheduleItemRequest>(req => req.IsRemote == true)), Times.Once);
+     
+        }
+
+        [Fact]
+        public async Task CreateScheduleItemRequest_WhenIsRemoteIsFalse_ShouldSetIsRemoteToFalse()
+        {
+            // Arrange
+            var scheduleItem = new ScheduleItem { Id = 1, Ad = new Ad { TutorId = 2 } };
+            SetupMockGetScheduleItemById(scheduleItem);
+
+            var request = new CreateScheduleItemRequestRequest
+            {
+                StudentId = 2,
+                ScheduleItemId = 1,
+                IsRemote = false
+            };
+
+            // Act
+            await StudentService.CreateScheduleItemRequest(request);
+
+            // Assert
+            MockScheduleItemRequestRepository.Verify(repo => repo.Create(It.Is<ScheduleItemRequest>(req => req.IsRemote == false)), Times.Once);
+        }
+
+        [Fact]
+        public async Task CreateScheduleItemRequest_WhenRequestSent_ShouldReturnSuccess()
+        {
+            // Arrange
+            var scheduleItem = new ScheduleItem { Id = 1 };
+            var studentId = 1;
+            var isRemote = true;
+
+            SetupMockGetScheduleItemById(scheduleItem);
+
+            var request = new CreateScheduleItemRequestRequest
+            {
+                StudentId = studentId,
+                ScheduleItemId = scheduleItem.Id,
+                IsRemote = isRemote
+            };
+
+            // Act
+            var response = await StudentService.CreateScheduleItemRequest(request);
+
+            // Assert
+            Assert.True(response.Success);
+            Assert.NotNull(response.CreatedScheduleItemRequestId);
```

Comment:
VS shows me a green squiggly line here and I agree. Don't check if `int` is not null. Value types are unable to be null (would not compile if you tried that), unless explicitly declared as nullable - `int?` is a a different type than `int`.